modules/LocationUtils.js

```diff
@@ -74,3 +74,7 @@ export const locationsAreEqual = (a, b) =>
   a.hash === b.hash &&
   a.key === b.key &&
   valueEqual(a.state, b.state);
+
+export const shouldReplace = (location, nextPath, history, nextState) => 
```

Comment:
I find this function a bit hard to read. Especially with the mixed parameters.

Just thinking out loud here, I could be very wrong, but what if you joined the params?

```js
shouldReplace(currentLocationAndState, newPathAndState) { ... }

...

if(shouldReplace(
  { location: history.location, state: history.state },
  { path, state }
)) {
  replace(path, state);
} else {
  push(path, state)
}
```

Reply:
I think `link` method is easy to understand now. Not the same with `shouldReplace` method, so I have made some changes in order to be easier to understand it. Tell me what you think.